internal/checker/checker.go

```diff
@@ -541,19 +367,25 @@ func (c *Checker) Structure(ctx context.Context, errChan chan<- error) {
 		}
 	}
 
-	treeIDChan := make(chan restic.ID)
-	treeJobChan1 := make(chan treeJob)
-	treeJobChan2 := make(chan treeJob)
+	wg, ctx := errgroup.WithContext(ctx)
+	treeStream := restic.StreamTrees(ctx, wg, c.repo, trees, func(treeID restic.ID) bool {
+		c.blobRefs.Lock()
```

Comment:
Here we have the same situation that it may be hard to see that this lock is needed because of the concurrent use in `checkTree` - maybe also add a comment here?

Reply:
I've added a small comment